src/main/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessor.java

```diff
@@ -0,0 +1,249 @@
+                    offsetBy, PAGE_SIZE, studyId);
+
+            csvFile = fileHelper.newFile(tmpDir, CSV_FILE_NAME);
+
+            try (CSVWriter csvFileWriter = new CSVWriter(fileHelper.getWriter(csvFile))) {
+                writeAccountSummaries(csvFileWriter, accountSummaries, offsetBy, appId, orgMembership, studyId);
+            } catch (IOException ex) {
+                throw new WorkerException("Error creating file " + csvFile + ": " + ex.getMessage(), ex);
+            }
+
+            // zip up and email the csv file
+            AppInfo appInfo = dynamoHelper.getApp(appId);
+            AccountInfo accountInfo = bridgeHelper.getAccountInfo(appId, userId);;
+            zipFile = fileHelper.newFile(tmpDir, ZIP_FILE_NAME);
+
+            zipFiles(csvFile, zipFile, password);
+            sesHelper.sendEmailWithAttachmentToAccount(appInfo, accountInfo, zipFile.getAbsolutePath());
+
+        } catch (BridgeSDKException ex) {
+            int status = ex.getStatusCode();
+            if (status >= 400 && status < 500) {
+                throw new PollSqsWorkerBadRequestException(ex);
+            } else {
+                throw new RuntimeException(ex);
+            }
+        } finally {
+            cleanupFiles(csvFile, zipFile, tmpDir);
+            LOG.info("request took " + requestStopwatch.elapsed(TimeUnit.SECONDS) +
+                    " seconds for userId =" + userId + ", app=" + appId);
+        }
+    }
+
+    /** Write all of the account summaries to the CSV file */
+    private void writeAccountSummaries(CSVWriter csvFileWriter, List<AccountSummary> accountSummaries, int offsetBy,
+                                       String appId, String orgMembership, String studyId) throws IOException, InterruptedException {
+        // write csv headers
+        String[] csvHeaders = getCsvHeaders(accountSummaries.get(0));
+        csvFileWriter.writeNext(csvHeaders);
+
+        // write the rest of the account summaries
+        while (!accountSummaries.isEmpty()) {
+            for (AccountSummary accountSummary : accountSummaries) {
+                String[] row = getAccountSummaryArray(accountSummary, csvHeaders);
+                csvFileWriter.writeNext(row);
+            }
+            // get next set of account summaries
+            offsetBy += PAGE_SIZE;
+            accountSummaries = bridgeHelper.getAccountSummariesForApp(appId, orgMembership, offsetBy, 0, studyId);
+
+            // avoid burning out Bridge Server
+            Thread.sleep(THREAD_SLEEP_INTERVAL);
+        }
+    }
+
+    /** Given an Object, return a String array of all of its attribute keys
+     *  Visible for unit testing
+     */
+    String[] getCsvHeaders(Object accountSummary) {
+        JsonElement json = RestUtils.toJSON(accountSummary);
+        Set<String> keys = json.getAsJsonObject().keySet();
+        List<String> list = new ArrayList<>(keys);
+        return list.toArray(new String[0]);
+    }
```

Comment:
It looks like this might not work, because if an account doesn't have an attribute, it'll omit the field instead of providing that field with a null result. We may need to use Reflection instead to get all fields, or have a configurable list of fields